stl/src/primitives.hpp

```diff
@@ -43,18 +44,8 @@ namespace Concurrency {
             CONDITION_VARIABLE m_condition_variable = CONDITION_VARIABLE_INIT;
         };
 
+        [[nodiscard]] inline stl_condition_variable_win7* _Get_cond_var(::_Cnd_internal_imp_t* _Cond) noexcept {

```

Comment:
We should really get rid of the `Concurrency::details` namespace. (Obviously not in this PR.) It's a historical artifact of previous dependence on ConCRT which is now just weirdly confusing for new readers. Maybe a "Good First Issue" issue?

Reply:
That would be *much* more spicy :hot_pepper: :fire: than I'd want to mark as a "good first issue", because some symbols are exported:

```
D:\GitHub\STL\out\x64\out\bin\amd64>dumpbin /exports msvcp140d_oss.dll | rg Concurrency::details
        404  193 00060320 ?CaptureCallstack@platform@details@Concurrency@@YA_KPEAPEAX_K1@Z = ?CaptureCallstack@platform@details@Concurrency@@YA_KPEAPEAX_K1@Z (unsigned __int64 __cdecl Concurrency::details::platform::CaptureCallstack(void * *,unsigned __int64,unsigned __int64))
        405  194 00060360 ?GetCurrentThreadId@platform@details@Concurrency@@YAJXZ = ?GetCurrentThreadId@platform@details@Concurrency@@YAJXZ (long __cdecl Concurrency::details::platform::GetCurrentThreadId(void))
        406  195 00060370 ?GetNextAsyncId@platform@details@Concurrency@@YAIXZ = ?GetNextAsyncId@platform@details@Concurrency@@YAIXZ (unsigned int __cdecl Concurrency::details::platform::GetNextAsyncId(void))
        407  196 00060200 ?ReportUnhandledError@_ExceptionHolder@details@Concurrency@@AEAAXXZ = ?ReportUnhandledError@_ExceptionHolder@details@Concurrency@@AEAAXXZ (private: void __cdecl Concurrency::details::_ExceptionHolder::ReportUnhandledError(void))
        411  19A 00060390 ?_Assign@_ContextCallback@details@Concurrency@@AEAAXPEAX@Z = ?_Assign@_ContextCallback@details@Concurrency@@AEAAXPEAX@Z (private: void __cdecl Concurrency::details::_ContextCallback::_Assign(void *))
        417  1A0 000603A0 ?_CallInContext@_ContextCallback@details@Concurrency@@QEBAXV?$function@$$A6AXXZ@std@@_N@Z = ?_CallInContext@_ContextCallback@details@Concurrency@@QEBAXV?$function@$$A6AXXZ@std@@_N@Z (public: void __cdecl Concurrency::details::_ContextCallback::_CallInContext(class std::function<void __cdecl(void)>,bool)const )
        419  1A2 000603D0 ?_Capture@_ContextCallback@details@Concurrency@@AEAAXXZ = ?_Capture@_ContextCallback@details@Concurrency@@AEAAXXZ (private: void __cdecl Concurrency::details::_ContextCallback::_Capture(void))
        552  227 00060440 ?_IsCurrentOriginSTA@_ContextCallback@details@Concurrency@@CA_NXZ = ?_IsCurrentOriginSTA@_ContextCallback@details@Concurrency@@CA_NXZ (private: static bool __cdecl Concurrency::details::_ContextCallback::_IsCurrentOriginSTA(void))
        553  228 00060450 ?_IsNonBlockingThread@_Task_impl_base@details@Concurrency@@SA_NXZ = ?_IsNonBlockingThread@_Task_impl_base@details@Concurrency@@SA_NXZ (public: static bool __cdecl Concurrency::details::_Task_impl_base::_IsNonBlockingThread(void))
        569  238 000604A0 ?_LogCancelTask@_TaskEventLogger@details@Concurrency@@QEAAXXZ = ?_LogCancelTask@_TaskEventLogger@details@Concurrency@@QEAAXXZ (public: void __cdecl Concurrency::details::_TaskEventLogger::_LogCancelTask(void))
        570  239 000604B0 ?_LogScheduleTask@_TaskEventLogger@details@Concurrency@@QEAAX_N@Z = ?_LogScheduleTask@_TaskEventLogger@details@Concurrency@@QEAAX_N@Z (public: void __cdecl Concurrency::details::_TaskEventLogger::_LogScheduleTask(bool))
        571  23A 000604C0 ?_LogTaskCompleted@_TaskEventLogger@details@Concurrency@@QEAAXXZ = ?_LogTaskCompleted@_TaskEventLogger@details@Concurrency@@QEAAXXZ (public: void __cdecl Concurrency::details::_TaskEventLogger::_LogTaskCompleted(void))
        572  23B 000604D0 ?_LogTaskExecutionCompleted@_TaskEventLogger@details@Concurrency@@QEAAXXZ = ?_LogTaskExecutionCompleted@_TaskEventLogger@details@Concurrency@@QEAAXXZ (public: void __cdecl Concurrency::details::_TaskEventLogger::_LogTaskExecutionCompleted(void))
        573  23C 000604E0 ?_LogWorkItemCompleted@_TaskEventLogger@details@Concurrency@@QEAAXXZ = ?_LogWorkItemCompleted@_TaskEventLogger@details@Concurrency@@QEAAXXZ (public: void __cdecl Concurrency::details::_TaskEventLogger::_LogWorkItemCompleted(void))
        574  23D 000604F0 ?_LogWorkItemStarted@_TaskEventLogger@details@Concurrency@@QEAAXXZ = ?_LogWorkItemStarted@_TaskEventLogger@details@Concurrency@@QEAAXXZ (public: void __cdecl Concurrency::details::_TaskEventLogger::_LogWorkItemStarted(void))
        615  266 00060B80 ?_Release_chore@details@Concurrency@@YAXPEAU_Threadpool_chore@12@@Z = ?_Release_chore@details@Concurrency@@YAXPEAU_Threadpool_chore@12@@Z (void __cdecl Concurrency::details::_Release_chore(struct Concurrency::details::_Threadpool_chore *))
        619  26A 00060500 ?_ReportUnobservedException@details@Concurrency@@YAXXZ = ?_ReportUnobservedException@details@Concurrency@@YAXXZ (void __cdecl Concurrency::details::_ReportUnobservedException(void))
        620  26B 00060BC0 ?_Reschedule_chore@details@Concurrency@@YAHPEBU_Threadpool_chore@12@@Z = ?_Reschedule_chore@details@Concurrency@@YAHPEBU_Threadpool_chore@12@@Z (int __cdecl Concurrency::details::_Reschedule_chore(struct Concurrency::details::_Threadpool_chore const *))
        621  26C 00060530 ?_Reset@_ContextCallback@details@Concurrency@@AEAAXXZ = ?_Reset@_ContextCallback@details@Concurrency@@AEAAXXZ (private: void __cdecl Concurrency::details::_ContextCallback::_Reset(void))
        624  26F 00060C40 ?_Schedule_chore@details@Concurrency@@YAHPEAU_Threadpool_chore@12@@Z = ?_Schedule_chore@details@Concurrency@@YAHPEAU_Threadpool_chore@12@@Z (int __cdecl Concurrency::details::_Schedule_chore(struct Concurrency::details::_Threadpool_chore *))
```

I'll add a personal todo to look into what we can change, though. Thanks!

Reply:
That is _quite_ a few "pretty" exports that I wasn't aware of! That's probably not a good second or even third issue; it may have to wait for vNext.